pkg/k8s/watchers/watcher.go

```diff
@@ -928,7 +934,35 @@ func (k *K8sWatcher) K8sEventReceived(apiResourceName, scope, action string, val
 	k.k8sResourceSynced.SetEventTimestamp(apiResourceName)
 }
 
+// GetIndexer returns an index to a k8s cache store for the given resource name.
+// Objects gotten using returned stores should *not* be mutated as they
+// are references to internal k8s watcher store state.
+func (k *K8sWatcher) GetIndexer(name string) cache.Indexer {
+	switch name {
+	case "ciliumendpointslice":
+		k.ciliumEndpointSliceStoreMU.Lock()
+		defer k.ciliumEndpointSliceStoreMU.Unlock()
+		return k.ciliumEndpointSliceStore
+	default:
+		return nil
+	}
+}
+
+// SetStore lets you set a named cache store, only used for testing.
+func (k *K8sWatcher) SetIndexer(name string, indexer cache.Indexer) {
+	switch name {
+	case "ciliumendpointslice":
```

Comment:
I would suggest to `panic` in the default case since that is a developer's mistake.